### PR TITLE
Added Context Menu for links

### PR DESCRIPTION
**Pull request by Clubfan22**
Long-pressing on a link button now shows a context menu with following actions:
* Copy URL to clipboard
* Share URL
* Open URL in external browser

In case a link is an image, two additional actions are available:
* Save Image
* Share Image

The actions are configurable through the menu preferences.

I moved some image-and-permission-related code from RedditPreparedPost into own classes to make it better reusable.

### CODE REVIEW

**Comment by Clubfan22** [comment]
I don't know why the Travis build fails, everything builds fine on my windows machine.
It might be related to the Travis problems though: https://www.traviscistatus.com/

File: src/main/java/org/quantumbadger/redreader/image/SaveImageCallback.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
+package org.quantumbadger.redreader.image;
```
Comment (QuantumBadger) [inline src/main/java/org/quantumbadger/redreader/image/SaveImageCallback.java]: It looks like the new files should have a GPL notice at the top :)

**Comment by QuantumBadger** [comment]
Thanks, this is excellent stuff! I've left one comment above that new files need a GPL notice (which can just be copied from one of the other files), but other than that this looks ready to merge in!

It looks like the German translation additions have been included in this PR as well, but that's fine. In future, keeping them in separate branches should avoid the conflict.

**Comment by Clubfan22** [comment]
Sorry for missing the copyright headers, didn't notice it with IntelliJ's auto-collapse and thought it was set  up automatically, just added those. And yes, I accidentally started this branch off the the localizastion changes, won't happen next time :)

**Comment by QuantumBadger** [comment]
Great, thanks! I'll merge this in now.

**Pull request merged**